AlpsSDK/NotificationManager.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  NotificationManager.swift
+//  AlpsSDK
+//
+//  Created by Wen on 13.10.17.
+//  Copyright © 2017 Alps. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import UserNotifications
+
+class NotificationManager : NSObject, UNUserNotificationCenterDelegate {
+    var alpsManager : AlpsManager
+    var deviceToken : Data? = nil
+    var deviceTokenString : String = ""
+    var seenError : Bool = false
+    
+    init(alpsManager: AlpsManager){
+        self.alpsManager = alpsManager
+        super.init()
+    }
+    
+    func initialize() {
+        self.seenError = registerForPushNotifications()
+    }
+    
+    private func registerForPushNotifications() -> Bool{
+        // iOS 10 support
+        if #available(iOS 10, *) {
+            UNUserNotificationCenter.current().requestAuthorization(options:[.badge, .alert, .sound]){ (granted, error) in
+                print("***********************************************")
+                print("Permission granted: \(granted)")
+                
+                guard granted else {self.seenError = true
+                    return }
+                // If get notification settings is not .authorized, register for remote notification is not called
+                self.getNotificationSettings()
+                UNUserNotificationCenter.current().delegate = self
+                UIApplication.shared.registerForRemoteNotifications()
+            }
+        }
+            // iOS 9 support
+        else if #available(iOS 9, *) {
+            UIApplication.shared.registerUserNotificationSettings(UIUserNotificationSettings(types: [.badge, .sound, .alert], categories: nil))
+            self.getNotificationSettings()
+            UIApplication.shared.registerForRemoteNotifications()
+        }
+            // iOS 8 support
```

Comment:
I think at some point, we should stop supporting.

Also note that 8 and 9 have exactly the same code.

iOS9+ in most cases is over 99% now.
https://david-smith.org/iosversionstats/

Reply:
Ok ! Yes I also agree with stop supporting some ios x previous version. But wanted to add the most and correct later 👍

Reply:
@tharpa what version we need to support?